homeassistant/components/fritz/binary_sensor.py

```diff
@@ -17,79 +24,94 @@
+        async_add_entities(entities, True)
 
-class FritzBoxConnectivitySensor(FritzBoxBaseEntity, BinarySensorEntity):
+
+class FritzBoxBinarySensor(FritzBoxBaseEntity, BinarySensorEntity):
     """Define FRITZ!Box connectivity class."""
 
     def __init__(
-        self, fritzbox_tools: FritzBoxTools, device_friendly_name: str
+        self,
+        fritzbox_tools: FritzBoxTools,
+        device_friendly_name: str,
+        sensor_type: BinarySensorEntityDescription,
     ) -> None:
         """Init FRITZ!Box connectivity class."""
-        self._unique_id = f"{fritzbox_tools.unique_id}-connectivity"
-        self._name = f"{device_friendly_name} Connectivity"
-        self._is_on = True
-        self._is_available = True
+        self.entity_description: BinarySensorEntityDescription = sensor_type
+        self._attr_device_class = self.entity_description.device_class
+        self._attr_name = f"{device_friendly_name} {self.entity_description.name}"
+        self._attr_unique_id = (
+            f"{fritzbox_tools.unique_id}-{self.entity_description.key}"
+        )
         super().__init__(fritzbox_tools, device_friendly_name)
 
-    @property
-    def name(self) -> str:
-        """Return name."""
-        return self._name
-
-    @property
-    def device_class(self) -> str:
-        """Return device class."""
-        return DEVICE_CLASS_CONNECTIVITY
-
-    @property
-    def is_on(self) -> bool:
-        """Return status."""
-        return self._is_on
-
-    @property
-    def unique_id(self) -> str:
-        """Return unique id."""
-        return self._unique_id
-
-    @property
-    def available(self) -> bool:
-        """Return availability."""
-        return self._is_available
-
     def update(self) -> None:
         """Update data."""
         _LOGGER.debug("Updating FRITZ!Box binary sensors")
-        self._is_on = True
-        try:
-            if (
-                self._fritzbox_tools.connection
-                and "WANCommonInterfaceConfig1"
-                in self._fritzbox_tools.connection.services
-            ):
-                link_props = self._fritzbox_tools.connection.call_action(
-                    "WANCommonInterfaceConfig1", "GetCommonLinkProperties"
-                )
-                is_up = link_props["NewPhysicalLinkStatus"]
-                self._is_on = is_up == "Up"
-            else:
-                if self._fritzbox_tools.fritz_status:
-                    self._is_on = self._fritzbox_tools.fritz_status.is_connected
-
-            self._is_available = True
 
+        try:
+            status: FritzStatus = self._fritzbox_tools.fritz_status
+            userinferface_x_avm: dict[
+                str, Any
+            ] = self._fritzbox_tools.connection.call_action(
+                "UserInterface", "X_AVM-DE_GetInfo"
+            )
+            userinferface1: dict[
+                str, Any
+            ] = self._fritzbox_tools.connection.call_action("UserInterface1", "GetInfo")
```

Comment:
These two requests are only necessary for the firmware update sensor, so I'd recommend avoid doing them when not needed (especially considering this polling is not using DataUpdateCoordinator, which might be worth exploring, too).

Reply:
With switch to DataUpdateCoordinator in future, but until then those 2 calls are needed to get the info needed.
I can just add a if on the second one after checking from the first if a new firmware is available.

Reply:
Definitely try to reduce API calls, so let's do that. Or maybe we should wait with update available sensor until DataUpdateCoordinator is there ?

Reply:
Actually, this should use a data update coordinator. All 3 sensors will now fetch the same data, that's not good.

Reply:
Updated approach, should work for now